File: src/main/java/de/byjoker/myjfql/user/UserService.java
package de.byjoker.myjfql.user;

import de.byjoker.myjfql.util.StorageService;

import java.util.List;

public interface UserService extends StorageService {

    void createUser(User user);

    void saveUser(User user);

    User getUserByIdentifier(String identifier);

    User getUserByName(String name);

    User getUser(String id);

    boolean existsUserByIdentifier(String identifier);

    boolean existsUserByName(String name);

    boolean existsUser(String id);

    void deleteUser(String id);

    List<User> getUsers();

}

File: src/main/java/de/byjoker/myjfql/user/UserService.kt
package de.byjoker.myjfql.user

interface UserService {

    fun createUser(user: User)
    fun saveUser(user: User)
    fun deleteUser(id: String)
    fun load(table: UsersTable)
    fun getUser(id: String): User?
    fun getUserByName(name: String): User?
    fun getUserByIdentifier(identifier: String): User?
    fun getUsers(): List<User>

}
